workflow-runtime/src/test/java/com/squareup/workflow1/RenderWorkflowInTest.kt

```diff
@@ -151,6 +153,35 @@ class RenderWorkflowInTest {
     assertEquals("updated state", restoredRenderings.value.rendering.first)
   }
 
+  // https://github.com/square/workflow-kotlin/issues/223
+  @Test fun `snapshots are lazy`() {
+    lateinit var sink: Sink<String>
+    var snapped = false
+
+    val workflow = Workflow.stateful<Unit, String, Nothing, String>(
+        initialState = { _, _ -> "unchanging state" },
+        snapshot = { Snapshot.of {
+          // We need to use snapped b/c the fail call is swallowed.
+          // https://github.com/square/workflow-kotlin/issues/223
+          snapped = true
+          fail("Snapshot must not be resolved.")
+        } },
+        render = { _, state ->
+          sink = actionSink.contraMap { action { this.state = it } }
+          state
+        }
+    )
+    val props = MutableStateFlow(Unit)
+    val renderings = renderWorkflowIn(workflow, scope, props) {}
+
+    renderings.value.rendering.let { rendered ->
+      assertEquals("unchanging state", rendered)
+      sink.send("unchanging state")
+    }
+
+    assertFalse(snapped)
```

Comment:
I'd probably add one more check here to make sure the next snapshot is not equal, and that the first snapshot is still not snapped.